### PR TITLE
ELO snippet formatting + performance

### PR DESCRIPTION
**Pull request by OutThisLife**
Not assigning `arr.length` to a variable in a for loop means that the read is done on every iteration.

## Description

**Resolves** #(issue number) 

## What does your PR belong to?
- [x] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)
- [ ] Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking improvement of a snippet)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by Chalarangelo** [comment]
Thanks for taking the time to contribute!

Not sure if the performance enhancement is significant enough to be worth the changes (which could be argued to harm readability just a tiny bit). Firefox performance shows a different of about 1%, whereas Chrome has already optimized this in its engine, so there is absolutely no performance difference. 

JSPerf for this PR can be found [here](https://jsperf.com/elo-comparsion).

I'll mark as `opinions needed`, see what others think.

**Comment by skatcat31** [comment]
There's other readability we can address on this question before that... These changes do bring in the proper placing and return structure of a guard statement(if statement that short circuits) meaning the else can be removed.

The 'being less readable' though is a little of a misnomer. For loops are anything BUT readable. `(Decleration; Comparison; Step)` is a footprint that will always be... awful. It's why you see in a LOT of places people taking declaration out ENTIRELY before the for loop and using footprints like `(; index < length; index++)` or using do...while loops before the days of the Array methods.

So my opinion is that it's already unreadable... I do see a need to rename l as length though so it's not lost what it is... possibly change it to index and jndex(common inside jokes) if we want to improve readability.

After all readable programs read like stories. It's why promises and composition are so popular:

Promise.resolve(compare(a,b)).then(winner_takes_all)

So in the end, declaring it can help redability. It probably just needs to be fleshed out a bit so that we don't get coded confusing code like

```js
return i < l && a < b ? a : r
```

**Comment by skatcat31** [comment]
Example of making it more readable:

```js
for (let index = 0, length = ratings.length; index < length; index++) {
  let jndex = index;
 
  while (jndex < length - 1) {
    [ratings[iindex], ratings[jndex + 1]] = elo([ratings[index], ratings[jndex + 1]], kFactor);
    jndex++;
  }
}
```

This is eschewing more updated coding practices to illustrate a point about readability being relative

File: snippets/elo.md [hunk 1]
```diff
@@ -15,17 +15,20 @@ const elo = ([...ratings], kFactor = 32, selfRating) => {
...
-  } else {
-    for (let i = 0; i < ratings.length; i++) {
-      let j = i;
-      while (j < ratings.length - 1) {
-        [ratings[i], ratings[j + 1]] = elo([ratings[i], ratings[j + 1]], kFactor);
-        j++;
-      }
+  } 
+  
+  for (let i = 0, l = ratings.length; i < l; i++) {
```
Comment (skatcat31) [inline snippets/elo.md]: While it leads to a speed improvements I think we do to much short cutting of names into abstracts here. We shoudl probably expand the names to make this more readable than a bunch of single letters

File: snippets/elo.md [hunk 2]
```diff
@@ -15,17 +15,20 @@ const elo = ([...ratings], kFactor = 32, selfRating) => {
...
+    let j = i;
+    
+    while (j < l - 1) {
+      [ratings[i], ratings[j + 1]] = elo([ratings[i], ratings[j + 1]], kFactor);
+      j++;
```
Comment (skatcat31) [inline snippets/elo.md]: If we're constantly using j+1 we should just `j++` before we execute this code or `++j` on the first call so we don't have to constantly put j+1

File: snippets/elo.md (hunk 1 above)
Comment (Chalarangelo) [reply snippets/elo.md]: `i` and `j` as iterators are fine (they are also part of the guidelines). `len` or `length` for the length should be a bit better than the current name.

File: snippets/elo.md (hunk 2 above)
Comment (Chalarangelo) [reply snippets/elo.md]: Yeah, `j++` seems logical.

Comment (fejes713) [reply snippets/elo.md]: > There's more that can be done in terms of readability, but a modest speed improvement and a proper guard statement is appreciated.

I agree with this one. Speaking about readability this snippet is categorized as advanced as far as I remember so readability was always a bit tough here. We assume that more experienced devs are going to read these anyways so ... Let's leave readability to some other PR like @skatcat31  mentioned.

Thanks for contributing!

**Pull request merged**